File: tests/NumberLiterals.cs
namespace Demo {
    public partial class NumberLiterals {
        public const int HexVal = 0x1A;
        public const int BinVal = 0b1011;
        public int Value() {
            int result;
            result = HexVal + BinVal;
            return result;
        }
    }
}

File: tests/NumberLiterals.pas
namespace Demo;

type
  NumberLiterals = public class
  public
    const HexVal: Integer = $1A;
    const BinVal: Integer = %1011;
    function Value: Integer;
  end;

implementation

function NumberLiterals.Value: Integer;
begin
  result := HexVal + BinVal;
end;

end.
